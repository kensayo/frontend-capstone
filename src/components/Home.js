import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { fetchComputers } from '../redux/computers/computers';
import './assets/home.css';

const Home = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchComputers());
  }, []);

  const computersState = useSelector((state) => state.computersReducer.computers);

  return (
    <div className="home m-4">
      <nav className="home-nav m-sd-4">
        <div />
        <h1 className="pt-md-5 mt-md-5">Computers</h1>
        <div>
          <i className="fa fa-search align-middle" aria-hidden="true" />
        </div>
      </nav>
      <div className="computers-container pt-4">
        { computersState.map((computer) => (
          <div className=" computer-item p-2" key={computer.id}>
            <Link to={`/computer/${computer.id}`} style={{ textDecoration: 'none', color: '#212529' }}>
              <div className="card m-2">
                <div className="card-body">
                  <div className="img-background" style={{ backgroundImage: `url(${computer.image})` }} />
                  <div className="computer-information pt-4">
                    <div className="d-flex flex-column p-3 computer-info">
                      <h1>{` Computer: ${computer.brand}`}</h1>
                      <br />
                      <p>{` Hdd: ${computer.hdd} TB` }</p>
                      <p>{` Processor: ${computer.processor}`}</p>
                      <p>{` Ram: ${computer.ram}`}</p>
                      <p>{`Price: $${computer.price}`}</p>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
            </Link>
          </div>
        ))}

      </div>
    </div>
  );
};
export default Home;
