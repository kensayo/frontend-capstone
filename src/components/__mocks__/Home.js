import PropTypes from 'prop-types';

const Home = (props) => {
  const {
    brand, hdd, ram, price, image,
  } = props;
  return (
    <div className="home m-4">
      <nav className="home-nav m-sd-4">
        <div />
        <h1 className="pt-md-5 mt-md-5">Computers</h1>
        <div>
          <i className="fa fa-search align-middle" aria-hidden="true" />
        </div>
      </nav>
      <div className="computers-container pt-4">
        <div className="card m-2">
          <div className="card-body">
            <div className="img-background" style={{ backgroundImage: `url(${image})` }} />
            <div className="computer-information pt-4">
              <div className="d-flex flex-column p-3 computer-info">
                <h1>{` Computer: ${brand}`}</h1>
                <br />
                <p>{` Hdd: ${hdd} TB` }</p>
                <p>{` Ram: ${ram}`}</p>
                <p>{`Price: $${price}`}</p>
              </div>
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Home;

Home.propTypes = {
  image: PropTypes.string.isRequired,
  hdd: PropTypes.string.isRequired,
  ram: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  brand: PropTypes.string.isRequired,
};
