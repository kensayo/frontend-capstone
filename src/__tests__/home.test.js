import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import Home from '../components/Home';
import store from '../redux/configureStore';

test('matches home component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
