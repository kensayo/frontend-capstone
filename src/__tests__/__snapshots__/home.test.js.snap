// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches home component snapshot 1`] = `
<div
  className="home m-4"
>
  <nav
    className="home-nav m-sd-4"
  >
    <div />
    <h1
      className="pt-md-5 mt-md-5"
    >
      Computers
    </h1>
    <div>
      <i
        aria-hidden="true"
        className="fa fa-search align-middle"
      />
    </div>
  </nav>
  <div
    className="computers-container pt-4"
  />
</div>
`;
